wallet/src/repository/wallet/mod.rs

```diff
@@ -380,16 +382,111 @@ where
 
         batch.put(keys::account_utxo_set(account), &new_utxo_set)?;
 
-        batch.put(keys::vtt(&transaction_hex_hash), &transaction)?;
+        batch.put(
+            keys::transaction(&transaction_hex_hash),
+            &types::Transaction::ValueTransfer(transaction.clone()),
+        )?;
+        batch.put(
+            keys::transaction_timestamp(account, transaction_id),
+            chrono::Local::now().timestamp(),
+        )?;
+        batch.put(keys::transaction_value(account, transaction_id), value)?;
+        batch.put(keys::transaction_fee(account, transaction_id), fee)?;
+        batch.put(
+            keys::transaction_entry(account, transaction_id),
+            model::TransactionEntry::Debit,
+        )?;
+        batch.put(
+            keys::transaction_type(account, transaction_id),
+            model::TransactionType::ValueTransfer,
+        )?;
+        if let Some(label) = label {
+            batch.put(keys::transaction_label(account, transaction_id), &label)?;
+        }
+        batch.put(keys::transactions_index(&transaction_hash), transaction_id)?;
+        batch.put(
+            keys::transaction_hash(account, transaction_id),
+            transaction_hash,
+        )?;
+
+        self.db.write(batch)?;
+
+        // update wallet state only after db has been updated
+        state.transaction_next_id = transaction_next_id;
+        state.utxo_set = new_utxo_set;
+
+        Ok(transaction)
+    }
+
+    /// Create a new data request transaction using available UTXOs.
+    pub fn create_data_req(
+        &self,
+        types::DataReqParams { label, request }: types::DataReqParams,
+    ) -> Result<types::DRTransaction> {
+        let fee = request.fee();
+        let value = request.total_witnesses_reward();
+        eprintln!("REQUEST: {:?}", request);
```

Comment:
Change eprintln to log or remove it.

```suggestion
        log::debug!("REQUEST: {:?}", request);
```